include/s3select_functions.h

```diff
@@ -643,6 +646,24 @@ struct _fn_substr : public base_function{
     
 };
 
+struct _fn_charlength : public base_function {
+
+    value v_str;
+    std::string a;
+ 
+    bool operator()(std::vector<base_statement*> * args,variable * result)
+    {
+        std::vector<base_statement*>::iterator iter = args->begin();
+        base_statement* str =  *iter;
+        v_str = str->eval();
+        assert(boost::conversion::try_lexical_convert<std::string>(v_str.str(), a) != false);
```

Comment:
as discussed , no need to validate its string content, but its string *type*
v_str.str() may contain invalid pointer.